### PR TITLE
Clean up more code

### PR DESCRIPTION
**Pull request by yujo-t**

### CODE REVIEW

File: group/assignment6/ParkingLot/Car.java [hunk 1]
```diff
@@ -2,34 +2,37 @@
 
 import java.util.Objects;
 
+/**
+ * Represents a car with an ID.
```
Comment (narciuch) [inline group/assignment6/ParkingLot/Car.java:6]: Represents a car with a unique ID.

File: group/assignment6/ParkingLot/Car.java [hunk 2]
```diff
@@ -2,34 +2,37 @@
...
     }
-
-    @Override
-    public int hashCode() {
-        return Objects.hash(id);
+    if (o == null || getClass() != o.getClass()) {
+      return false;
     }
+    Car car = (Car) o;
+    return id == car.id;
```
Comment (narciuch) [inline group/assignment6/ParkingLot/Car.java:24]: this.id == otherCar.id

to make it explicit

File: group/assignment6/ParkingLot/CarMove.java [hunk 3]
```diff
@@ -2,47 +2,53 @@
 
 import java.util.Objects;
 
+/**
+ * Represents a move of a car which specifies which space it is moving from and which space it is moving to.
```
Comment (narciuch) [inline group/assignment6/ParkingLot/CarMove.java:6]: Represents a move from a parking space to another by a car.

File: group/assignment6/ParkingLot/Move.java [hunk 4]
```diff
@@ -17,7 +17,7 @@ public boolean equals(Object o) {
     if (o == this) {
       return true;
     }
-    if (o == null || !(o instanceof Move)) {
+    if (o == null || getClass() != o.getClass()) {
```
Comment (narciuch) [inline group/assignment6/ParkingLot/Move.java:20]: Interesting, why has this been changed?

File: group/assignment6/ParkingLot/Space.java [hunk 5]
```diff
@@ -2,33 +2,39 @@
 
 import java.util.Objects;
 
+/**
+ * Represents a space with an ID.
```
Comment (narciuch) [inline group/assignment6/ParkingLot/Space.java:6]: Represents a car parking space with a unique ID.

**Review by narciuch — COMMENTED** [review]
LGTM

File: group/assignment6/ParkingLot/Move.java (hunk 4 above)
Comment (yujo-t) [reply group/assignment6/ParkingLot/Move.java:20]: For consistency

File: group/assignment6/ParkingLot/Car.java (hunk 1 above)
Comment (yujo-t) [reply group/assignment6/ParkingLot/Car.java:6]: This is not enforced within the class itself though so I don't think we should put this.

Comment (narciuch) [reply group/assignment6/ParkingLot/Car.java:6]: Ah yes, it is only enforced within the BiMap which uses the car. Good point

**Pull request merged**